FacebookWinFormsApp/FormMain.cs

```diff
@@ -694,17 +697,19 @@ private void fetchAlbumPictures(Album i_SelectedAlbum)
 
         private void buttonNostalgia_Click(object sender, EventArgs e)
         {
-            const bool v_ToBeVisible = true;
-            if (comboBoxMediaType.Text == k_ContentCategoryPhotos)
-            {
-                r_NostalgiaViewers[(int)eNostalgiaViewerIndex.PostViewer].SetVisibility(!v_ToBeVisible);
-                showNostalgiaPhoto();
-            }
-            else if (comboBoxMediaType.Text == k_ContentCategoryPosts)
-            {
-                r_NostalgiaViewers[(int)eNostalgiaViewerIndex.PhotoViewer].SetVisibility(!v_ToBeVisible);
-                showNostalgiaPost();
-            }
+            // const bool v_ToBeVisible = true;
+            // if (comboBoxMediaType.Text == k_ContentCategoryPhotos)
+            // {
+            //    r_NostalgiaViewers[(int)eNostalgiaViewerIndex.PostViewer].SetVisibility(!v_ToBeVisible);
+            //    showNostalgiaPhoto();
+            // }
+            // else if (comboBoxMediaType.Text == k_ContentCategoryPosts)
+            // {
+            //    r_NostalgiaViewers[(int)eNostalgiaViewerIndex.PhotoViewer].SetVisibility(!v_ToBeVisible);
+            //    showNostalgiaPost();
+            // }
+            SetRandomStrategy();
+            m_RandomStrategy.ShowRandomContent(r_LoggedInUser);
```

Comment:
You need to turn off the visibility of the other viewer when you show some viewer. In this case you can see that in lines 703 and 708 we've turned the other viewer off, but you haven't added this in your refactor and that may cause the bug you've mentioned.
I suggest adding the following method `hideViewers()` and invoke it after line 711:

```c#
private void hideViewers()
{
         const bool v_ToBeVisible = true;

         if (comboBoxMediaType.Text == k_ContentCategoryPhotos)
        {
            r_NostalgiaViewers[(int)eNostalgiaViewerIndex.PostViewer].SetVisibility(!v_ToBeVisible);
        
        }
        else if (comboBoxMediaType.Text == k_ContentCategoryPosts)
        {
            r_NostalgiaViewers[(int)eNostalgiaViewerIndex.PhotoViewer].SetVisibility(!v_ToBeVisible);
        }
}

```

Reply:
yep, that was it, it works now